src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java

```diff
@@ -79,14 +79,192 @@
+ * </p>
+ * <pre>
+ * class InputClassComplexity {
+ *   Set set = new HashSet(); // Set counted 1, HashSet ignored
+ *   Map map = new HashMap(); // Map counted 2, HashMap ignored
+ *   Date date = new Date(); // Counted, 3
+ *   Time time = new Time(); // Counted, 4
+ *   // mention of 16 other user defined classes
+ *   // Counted, 20
+ *   Space space = new Space(); // Violation
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to exclude classes with a regular expression
+ * {@code .*Reader$}:
+ * </p>
+ * <pre>
+ * &lt;module name="ClassFanOutComplexity"&gt;
+ *   &lt;property name="excludeClassesRegexps" value=".*Reader$"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Code examples:
+ * </p>
+ * <p>
+ * The check passes without violations in the following.
+ * </p>
+ * <pre>
+ * class InputClassComplexity {
+ *   Set set = new HashSet(); // Ignored due to default excludedClasses property
+ *   Map map = new HashMap(); // Ignored due to default excludedClasses property
+ *   Date date = new Date(); // Counted, 1
+ *   Time time = new Time(); // Counted, 2
+ *   // mention of 18 other user defined classes
+ *   // Counted, 20
+ *   BufferedReader br; // Ignored
+ * }
+ * </pre>
+ * <p>
+ * The check results in a violation in the following.
+ * </p>
+ * <pre>
+ * class InputClassComplexity {
+ *   Set set = new HashSet(); // Ignored due to default excludedClasses property
+ *   Map map = new HashMap(); // Ignored due to default excludedClasses property
+ *   Date date = new Date(); // Counted, 1
+ *   Time time = new Time(); // Counted, 2
+ *   // mention of 18 other user defined classes
+ *   // Counted, 20
+ *   File file; // Violation
+ * }
+ * </pre>
+ * <p>
+ * To configure the check with an excluded package {@code java.io}:
+ * </p>
+ * <pre>
+ * &lt;module name="ClassFanOutComplexity"&gt;
+ *   &lt;property name="excludedPackages" value="java.io"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Code examples:
+ * </p>
+ * <p>
+ * The check passes without violations in the following.
+ * </p>
+ * <pre>
+ * import java.io.BufferedReader;
+ *
+ * class InputClassComplexity {
+ *   Set set = new HashSet(); // Ignored due to default excludedClasses property
+ *   Map map = new HashMap(); // Ignored due to default excludedClasses property
+ *   Date date = new Date(); // Counted, 1
+ *   Time time = new Time(); // Counted, 2
+ *   // mention of 18 other user defined classes
+ *   // Counted, 20
+ *   BufferedReader br; // Ignored
+ * }
+ * </pre>
+ * <p>
+ * The check results in a violation in the following.
+ * </p>
+ * <pre>
+ * import java.util.StringTokenizer;
+ *
+ * class InputClassComplexity {
+ *   Set set = new HashSet(); // Ignored due to default excludedClasses property
+ *   Map map = new HashMap(); // Ignored due to default excludedClasses property
+ *   Date date = new Date(); // Counted, 1
+ *   Time time = new Time(); // Counted, 2
+ *   // mention of 18 other user defined classes
+ *   // Counted, 20
```

Comment:
please remove and update violation comment below

Reply:
Done